addons/stock/models/stock_move_line.py

```diff
@@ -249,18 +249,19 @@ def _apply_putaway_strategy(self):
         if self._context.get('avoid_putaway_rules'):
             return
         self = self.with_context(do_not_unreserve=True)
-        for package, smls in groupby(self, lambda sml: sml.result_package_id):
+        for (location_dest, package), smls in groupby(self, lambda sml: (sml.location_dest_id, sml.result_package_id)):
```

Comment:
Shouldn't it be grouped by `sml.move_id.location_dest_id` rather than `sml.location_dest_id`?